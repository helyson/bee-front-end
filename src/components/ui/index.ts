export { UIAvatar } from "./avatar/Avatar";
export { UIButton } from "./button/Button";
export { UIFormGroup } from "./form-group/FormGroup";
export { UIInput } from "./input/Input";
export { UILogo } from "./logo/Logo";
export { UIModal } from "./modal/Modal";
export { UISvgIcon } from "./icon/Icon";
export { UITable } from "./table/Table";
