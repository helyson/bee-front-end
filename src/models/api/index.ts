export { RequestEnum } from "./request.enum";
export * from "./status";
export * from "./request-api";
