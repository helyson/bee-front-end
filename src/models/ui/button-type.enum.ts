enum ButtonTypeEnum {
    Button = "btn",
    Link = "link",
    TextIcon= "textIcon"
  }
  
  export { ButtonTypeEnum};