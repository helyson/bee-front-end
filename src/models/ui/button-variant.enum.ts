enum ButtonVarianEnum {
  Primary = "primary",
  Secondary = "secondary",
  Gray = "gray",
}

export { ButtonVarianEnum };
