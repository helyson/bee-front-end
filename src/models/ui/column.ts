import { ReactNode } from "react";

interface Column<T> {
  key: string;
  label: string;
  width?: string
  render: (record: T, isHover?:boolean) => ReactNode;
}

export type { Column };
