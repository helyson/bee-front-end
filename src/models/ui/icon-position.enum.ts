enum IconPositionEnum {
  Left = "left",
  Right = "right",
}

export { IconPositionEnum };
