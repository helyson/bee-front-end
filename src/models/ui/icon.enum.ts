export enum IconEnum {
    Plus="plus",
    Close="close",
    Search="search",
    Back="back",
    Forward="forward"
  }