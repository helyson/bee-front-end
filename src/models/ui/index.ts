export { ButtonTypeEnum } from "./button-type.enum";
export { ButtonVarianEnum } from "./button-variant.enum";
export { IconEnum } from "./icon.enum";
export { IconPositionEnum } from "./icon-position.enum";
export * from "./column";
