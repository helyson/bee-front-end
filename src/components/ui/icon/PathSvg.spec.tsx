
import { IconEnum } from "src/models/ui";
import { getPath } from "./PathSvg";

describe(" getViewBox()", () => {
  test("should be render path Back", () => {
    const path = getPath(IconEnum.Back, "");
    expect(path).toMatchSnapshot();
  });
  test("should be render path Close", () => {
    const path = getPath(IconEnum.Close, "");
    expect(path).toMatchSnapshot();
  });
  test("should be render path Forward", () => {
    const path = getPath(IconEnum.Forward, "");
    expect(path).toMatchSnapshot();
  });
  test("should be render path Plus", () => {
    const path = getPath(IconEnum.Plus, "");
    expect(path).toMatchSnapshot();
  });
  test("should be render path Search", () => {
    const path = getPath(IconEnum.Search, "");
    expect(path).toMatchSnapshot();
  });
});