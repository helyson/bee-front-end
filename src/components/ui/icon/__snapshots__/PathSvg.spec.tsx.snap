// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` getViewBox() should be render path Back 1`] = `
<path
  d="M124.161,0C55.589,0,0,55.589,0,124.161s55.589,124.161,124.161,124.161c68.571,0,124.161-55.589,124.161-124.161   S192.732,0,124.161,0z M221.721,142.335H97.059l53.244,53.244l-28.284,28.284l-99.702-99.702l99.702-99.702l28.284,28.284   l-53.244,53.244h124.662V142.335z"
/>
`;

exports[` getViewBox() should be render path Close 1`] = `
<React.Fragment>
  <circle
    cx="14"
    cy="14"
    fill="#C4C4C4"
    r="14"
  />
  <path
    d="M8.17677 7.82323C8.65918 7.34082 9.44133 7.34082 9.92374 7.82323L19.8232 17.7227C20.3056 18.2051 20.3056 18.9873 19.8232 19.4697C19.3408 19.9521 18.5587 19.9521 18.0763 19.4697L8.17677 9.5702C7.69436 9.08779 7.69436 8.30564 8.17677 7.82323Z"
  />
  <path
    d="M19.8232 7.82323C20.3056 8.30564 20.3056 9.08779 19.8232 9.5702L9.92374 19.4697C9.44132 19.9521 8.65918 19.9521 8.17677 19.4697C7.69435 18.9873 7.69435 18.2051 8.17677 17.7227L18.0763 7.82323C18.5587 7.34082 19.3408 7.34082 19.8232 7.82323Z"
  />
</React.Fragment>
`;

exports[` getViewBox() should be render path Forward 1`] = `
<path
  d="M124.161,248.322c68.571,0,124.161-55.589,124.161-124.161S192.732,0,124.161,0C55.589,0,0,55.589,0,124.161   S55.589,248.322,124.161,248.322z M26.601,105.987h124.661L98.019,52.743l28.286-28.284l99.7,99.702l-99.7,99.702l-28.286-28.284   l53.243-53.244H26.601V105.987z"
/>
`;

exports[` getViewBox() should be render path Plus 1`] = `
<React.Fragment>
  <path
    d="M8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zM13 9h-4v4h-2v-4h-4v-2h4v-4h2v4h4v2z"
  />
</React.Fragment>
`;

exports[` getViewBox() should be render path Search 1`] = `
<path
  d="M54.734,9.053C39.12,18.067,27.95,32.624,23.284,50.039c-4.667,17.415-2.271,35.606,6.743,51.22  c12.023,20.823,34.441,33.759,58.508,33.759c7.599,0,15.139-1.308,22.287-3.818l30.364,52.592l21.65-12.5l-30.359-52.583  c10.255-8.774,17.638-20.411,21.207-33.73c4.666-17.415,2.27-35.605-6.744-51.22C134.918,12.936,112.499,0,88.433,0  C76.645,0,64.992,3.13,54.734,9.053z M125.29,46.259c5.676,9.831,7.184,21.285,4.246,32.25c-2.938,10.965-9.971,20.13-19.802,25.806  c-6.462,3.731-13.793,5.703-21.199,5.703c-15.163,0-29.286-8.146-36.857-21.259c-5.676-9.831-7.184-21.284-4.245-32.25  c2.938-10.965,9.971-20.13,19.802-25.807C73.696,26.972,81.027,25,88.433,25C103.597,25,117.719,33.146,125.29,46.259z"
/>
`;
